### PR TITLE
Feature - Added theme documentation for Box

### PR DESCRIPTION
**Pull request by MoiyadKayda**
## Screenshots:
image

## Checklist:

- [ ] I have updated the CHANGELOG with a summary of my changes
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/particles/box/documentation.stories.mdx [hunk 1]
```diff
@@ -76,4 +76,39 @@ export const Template = (args) => (
...
+    'border-color': '$colors.backgroundDark05',
+    'box-shadow': '0px 8px 8px -6px rgba(0,0,0,0.15)',
+    'margin': '0px 4px 12px 4px'
+  },
+  newVariant: {
+    'background-color': '$colors.backgroundDark10'
+  }
+}
+```
 TODO
```
Comment (krishan711) [inline src/particles/box/documentation.stories.mdx:114]: please remove this once the theming is added

Comment (MoiyadKayda) [reply src/particles/box/documentation.stories.mdx:107]: There is some problem with the progressCounter component, it has weird error that it is not able to access the PaddingSize enum.
And also when the example box theme is changed, the theme of the rest of the boxes also change, why it is so? it doesn't happen in the button example.

File: src/particles/box/themeExample.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export const BoxThemeExample = (): React.ReactElement => {
+  const defaultTheme = useTheme();
+  const [theme, setTheme] = React.useState<IBoxTheme>(defaultTheme.boxes.default);
+  const [themeEditorField, setThemeEditorField] = React.useState<string>('');
+  const [setThemeCallback, clearSetThemeCallback] = useDebouncedCallback(1000);
+  const currentThemeValue = React.useRef<IBoxTheme>(theme);
+
+  React.useEffect((): void => {
+    currentThemeValue.current = theme;
+    currentThemeValue.current['background-color'] = '$colors.brandPrimary';
```
Comment (MoiyadKayda) [inline src/particles/box/themeExample.tsx:24]: Is it because I am changing the value of background in useEffect

Comment (MoiyadKayda) [reply src/particles/box/themeExample.tsx:24]: if I remove this then there is no issue.

File: src/particles/box/themeExample.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { Stack } from '../../layouts';
+import { Direction } from '../../model';
+import { MultiLineInput } from '../../molecules';
+import { PaddingSize, Spacing } from '../../particles';
+import { useTheme } from '../../theming';
+import { mergeTheme } from '../../util';
+
+export const BoxThemeExample = (): React.ReactElement => {
+  const defaultTheme = useTheme();
+  // Note : I had to concat the default variant with the bordered one, so that the whole box theme is visible to the user
```
Comment (krishan711) [inline src/particles/box/themeExample.tsx:17]:
```suggestion
```

**Review by krishan711 — APPROVED** [review]

**Review by krishan711 — APPROVED** [review]

**Pull request merged**